src/datatable/io.jl

```diff
@@ -1,1198 +0,0 @@
-                   skipblanks::Bool = true,
-                   encoding::Symbol = :utf8,
-                   allowescapes::Bool = false,
-                   normalizenames::Bool = true)
-    # Open an IO stream based on pathname
-    # (1) Path is an HTTP or FTP URL
-    if startswith(pathname, "http://") || startswith(pathname, "ftp://")
-        error("URL retrieval not yet implemented")
-    # (2) Path is GZip file
-    elseif endswith(pathname, ".gz")
-        io = gzopen(pathname, "r")
-        nbytes = 2 * filesize(pathname)
-    # (3) Path is BZip2 file
-    elseif endswith(pathname, ".bz") || endswith(pathname, ".bz2")
-        error("BZip2 decompression not yet implemented")
-    # (4) Path is an uncompressed file
-    else
-        io = open(pathname, "r")
-        nbytes = filesize(pathname)
-    end
-
-    dt = try
-        readtable(io,
-                  nbytes,
-                  header = header,
-                  separator = separator,
-                  quotemark = quotemark,
-                  decimal = decimal,
-                  nastrings = nastrings,
-                  truestrings = truestrings,
-                  falsestrings = falsestrings,
-                  makefactors = makefactors,
-                  nrows = nrows,
-                  names = names,
-                  eltypes = eltypes,
-                  allowcomments = allowcomments,
-                  commentmark = commentmark,
-                  ignorepadding = ignorepadding,
-                  skipstart = skipstart,
-                  skiprows = skiprows,
-                  skipblanks = skipblanks,
-                  encoding = encoding,
-                  allowescapes = allowescapes,
-                  normalizenames = normalizenames)
-    finally
-        close(io)
-    end
-
-    return dt
-end
-
-"""
-    inlinetable(s[, flags]; args...)
-
-A helper function to process strings as tabular data for non-standard string
-literals. Parses the string `s` containing delimiter-separated tabular data
-(by default, comma-separated values) using `readtable`. The optional `flags`
-argument contains a list of flag characters, which, if present, are equivalent
-to supplying named arguments to `readtable` as follows:
-
-- `f`: `makefactors=true`, convert string columns to `CategoricalArray` columns
-- `c`: `allowcomments=true`, ignore lines beginning with `#`
-- `H`: `header=false`, do not interpret the first line as column names
-"""
-inlinetable(s::AbstractString; args...) = readtable(IOBuffer(s); args...)
-function inlinetable(s::AbstractString, flags::AbstractString; args...)
-    flagbindings = Dict(
-        'f' => (:makefactors, true),
-        'c' => (:allowcomments, true),
-        'H' => (:header, false) )
-    for f in flags
-        if haskey(flagbindings, f)
-            push!(args, flagbindings[f])
-        else
-            throw(ArgumentError("Unknown inlinetable flag: $f"))
-        end
-    end
-    readtable(IOBuffer(s); args...)
-end
-
-"""
```

Comment:
I'm a bit sad to see these non-standard string literals go. I wonder whether we could add them to CSV.jl. @quinnj?